lambdas/data-migration1/tests/test-rules.js

```diff
@@ -0,0 +1,399 @@
+      migrationDir,
+    },
+  });
+
+  await t.context.knex.migrate.latest();
+
+  await Promise.all([
+    putJsonS3Object(
+      process.env.system_bucket,
+      messageTemplateKey,
+      { meta: 'meta' }
+    ),
+    putJsonS3Object(
+      process.env.system_bucket,
+      workflowfile,
+      { testworkflow: 'workflow-config' }
+    ),
+  ]);
+});
+
+test.afterEach.always(async (t) => {
+  await t.context.knex('rules').del();
+  await t.context.knex('providers').del();
+  await t.context.knex('collections').del();
+});
+
+test.after.always(async (t) => {
+  await providersModel.deleteTable();
+  await collectionsModel.deleteTable();
+  await rulesModel.deleteTable();
+
+  await recursivelyDeleteS3Bucket(process.env.system_bucket);
+
+  await t.context.knex.destroy();
+  await t.context.knexAdmin.raw(`drop database if exists "${testDbName}"`);
+  await t.context.knexAdmin.destroy();
+});
+
+test.serial('migrateRuleRecord correctly migrates rule record', async (t) => {
+  const { knex, providerKmsKeyId } = t.context;
+  fakeCollection = fakeCollectionFactory();
+  fakeProvider = fakeProviderFactory({
+    encrypted: false,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  });
+  const fakeRule = generateFakeRule(fakeCollection.name, fakeCollection.version, fakeProvider.id);
+
+  await migrateCollectionRecord(fakeCollection, knex);
+  await migrateProviderRecord(fakeProvider, providerKmsKeyId, knex);
+  await migrateRuleRecord(fakeRule, knex);
+
+  const createdRecord = await t.context.knex.queryBuilder()
+    .select()
+    .table('rules')
+    .where({ name: fakeRule.name })
+    .first();
+
+  t.deepEqual(
+    omit(createdRecord, ['cumulus_id', 'collection_cumulus_id', 'provider_cumulus_id' ]),
+    omit(
+      {
+        name: fakeRule.name,
+        workflow: fakeRule.workflow,
+        meta: fakeRule.meta,
+        arn: fakeRule.rule.arn,
+        type: fakeRule.rule.type,
+        value: fakeRule.rule.value,
+        enabled: true,
+        log_event_arn: fakeRule.rule.logEventArn,
+        execution_name_prefix: null,
+        payload: fakeRule.payload,
+        queue_url: fakeRule.queueUrl,
+        tags: fakeRule.tags,
+        created_at: new Date(fakeRule.createdAt),
+        updated_at: new Date(fakeRule.updatedAt),
+      },
+      ruleOmitList
+    )
+  );
+});
+
+test.serial('migrateRuleRecord throws error on invalid source data from DynamoDb', async (t) => {
+  fakeCollection = fakeCollectionFactory();
+  fakeProvider = fakeProviderFactory({
+    encrypted: false,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  });
+  const fakeRule = generateFakeRule(fakeCollection.name, fakeCollection.version, fakeProvider.id);
+
+  // make source record invalid
+  delete fakeRule.workflow;
+
+  await t.throwsAsync(migrateRuleRecord(fakeRule, t.context.knex));
```

Comment:
Let's check the error type here to ensure it's the right type

```suggestion
  await t.throwsAsync(
    migrateRuleRecord(fakeRule, t.context.knex),
    { name: 'SchemaValidationError' }
  );
```